src/app/auth/auth.js

```diff
@@ -0,0 +1,152 @@
+angular.module('prx.auth', ['prx.ui.nav'])
+.config(function (PRXDrawerProvider) {
+  PRXDrawerProvider.register({
+    name: 'Sign In',
+    weight: PRXDrawerProvider.BOTTOM,
+    href: 'http://www.prx.org/sessions/new',
+    nav: true,
+    template: '<prx-auth-badge></prx-auth-badge>',
+    dropdownTemplate: '<prx-auth-window></prx-auth-window>'
+  });
+}).service('PrxAuth', function (ngHal, $q, $window) {
+  var PrxAuth = this,
+      deferred = $q.defer(),
+      currentUser = {loggedIn: false};
+  this.$$loginPromise = deferred.promise;
+
+  this.$checkLoggedIn = function () {
+    var iframe = angular.element('<iframe>'),
+        $body = angular.element($window.document.body),
+        uri = [FEAT.ID_SERVER + '/authorize?client_id=' + FEAT.ID_CLIENT_KEY],
+        nonce = [];
+    for (var i=0; i<8; i++) {
+      nonce.push(randomInt(0, 15).toString(16));
+    }
+    nonce = nonce.join('');
+    uri.push("&nonce=", nonce, "&response_type=token&prompt=none");
+    iframe.css('display', 'none');
+    iframe.attr('src', uri.join(''));
+
+    iframe.on('load', function onLoad() {
+      var query = false, accessToken;
+      try {
+        query = iframe[0].contentDocument.location.hash.replace(/^#/,'');
```

Comment:
`contentDocument` is a handle on the `window.document` of the page the iframe has loaded. We can't get a handle on it unless the origin is the same, so I have it wrapped in a try/catch here. If we are able to successfully pull the value (which should always be the case here, unless the client is misconfigured on id.prx) we pass it along.